python/mspasspy/db/schema.py

```diff
@@ -597,5 +597,83 @@ def set_writeable(self, key):
+    collection_name_list = []
+    if db not in schema_dic:
+        return collection_name_list
+    for c in schema_dic[db]:
+        collection_name_list.append(c)
+    return collection_name_list
+
 def _check_format(schema_dic):
-    pass
+    db_name_list = get_all_db_name(schema_dic)
+    collection_name_list = []
+    for db_name in db_name_list:
+        collection_name_list += get_all_collection_name_by_db(schema_dic, db_name)
+    
+    type_attribute_schema = schema.Schema({
+        'type':schema.And(str, lambda s: is_basic_type(s)),
+        schema.Optional('reference'):schema.And(str, lambda s: s in collection_name_list),
+        schema.Optional('concept'):str,
+        schema.Optional('aliases'):schema.Or(list, str),
+        schema.Optional('optional'):bool,
+        schema.Optional('readonly'):bool
+    }, ignore_extra_keys=True)
+    
+    reference_attribute_schema = schema.Schema({
+        'reference':schema.And(str, lambda s: s in collection_name_list),
+        schema.Optional('optional'):bool
+    }, ignore_extra_keys=True)
+    
+    metadata_attribute_schema = schema.Schema({
+        'collection':schema.And(str, lambda s: s in collection_name_list),
+        schema.Optional('readonly'):bool
+    }, ignore_extra_keys=True)
+    
+    database_collection_schema = schema.Schema({
+        schema.Optional('base'):schema.And(str, lambda s: s in collection_name_list),
+        'schema':schema.And(dict, schema.And(dict, lambda dic: ('_id' in dic and type_attribute_schema.validate(attr) 
+                                        if 'type' in dic[attr] else reference_attribute_schema.validate(attr) 
+                                        for attr in dic)))
+    }, ignore_extra_keys=True)
+    
+    metadata_collection_schema = schema.Schema({
+        'schema':schema.And(dict, lambda dic: (metadata_attribute_schema.validate(attribute) for attribute in dic))
+    }, ignore_extra_keys=True)
+    
+    database_schema = schema.Schema({
+        'wf_TimeSeries':schema.And(dict, lambda dic: database_collection_schema.validate(dic)),
+        'wf_Seismogram':schema.And(dict, lambda dic: database_collection_schema.validate(dic)),
+        schema.Optional('site'):schema.And(dict, lambda dic: database_collection_schema.validate(dic)),
+        schema.Optional('channel'):schema.And(dict, lambda dic: database_collection_schema.validate(dic)),
+        schema.Optional('source'):schema.And(dict, lambda dic: database_collection_schema.validate(dic)),
+        schema.Optional('history_object'):schema.And(dict, lambda dic: database_collection_schema.validate(dic)),
+        schema.Optional('elog'):schema.And(dict, lambda dic: database_collection_schema.validate(dic)),
+    }, ignore_extra_keys=True)
+    
+    metadata_schema = schema.Schema({
+        'TimeSeries':schema.And(dict, lambda dic: metadata_collection_schema.validate(dic)),
+        'Seismogram':schema.And(dict, lambda dic: metadata_collection_schema.validate(dic)),
```

Comment:
These two are OK for now, but it should be similar to the `Database` schema, so that we can support data types other than `TimeSeries` and `Seismogram` in the future.